### PR TITLE
Emit filter_deleted_event after filter is deleted

### PR DESCRIPTION
**Pull request by ehgmsdk20**
Changes:
- Emit filter_deleted_event after filter is deleted
- refactor event handler error

Related issues:
- https://github.com/ainblockchain/ain-js/issues/88

### CODE REVIEW

File: common/constants.js [hunk 1]
```diff
@@ -676,6 +676,8 @@ const ValueChangedEventSources = {
 const FilterDeletionReasons = {
   FILTER_TIMEOUT: 'FILTER_TIMEOUT',
   END_STATE_REACHED: 'END_STATE_REACHED',
+  DELETED_BY_USER: 'DELETED_BY_USER',
+  ERROR_OCCURED: 'ERROR_OCCURED',
```
Comment (platfowner) [inline common/constants.js:680]: OCCURRED?

File: event-handler/event-channel-manager.js [hunk 2]
```diff
@@ -52,6 +53,15 @@ class EventChannelManager {
     return channel;
   }
 
+  getChannelAndClientFilterIdByEventFilterId(eventFilterId) {
```
Comment (platfowner) [inline event-handler/event-channel-manager.js:56]: what about just adding getClientFilterIdFromEventFilterId() for client filter id and reuse getChannelByEventFilterId() for channel?

File: event-handler/state-event-tree-manager.js [hunk 3]
```diff
@@ -89,20 +89,22 @@ class StateEventTreeManager {
...
     if (!eventNode || !eventNode.filterIdSet) {
       throw new EventHandlerError(EventHandlerErrorCode.MISSING_FILTER_ID_SET,
-          `Can't find filterIdSet (eventNode: ${JSON.stringify(eventNode)})`, filterId);
+          `Can't find filterIdSet (eventNode: ${JSON.stringify(eventNode)})`);
     }
     if (!eventNode.filterIdSet.delete(filterId)) {
       throw new EventHandlerError(EventHandlerErrorCode.MISSING_FILTER_ID_IN_FILTER_ID_SET,
-          `Can't delete filter id from filterIdSet ` +
-          `(${JSON.stringify(eventNode.filterIdSet.values())})`, filterId);
+          `Can't delete filter id(${filterId}) from filterIdSet ` +
```
Comment (platfowner) [inline event-handler/state-event-tree-manager.js:96]: minor) space after 'id'?

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Thanks!

File: event-handler/event-channel-manager.js (hunk 2 above)
Comment (ehgmsdk20) [reply event-handler/event-channel-manager.js:56]: Oh, then i think we can use `getClientFilterIdFromGlobalFilterId()` in event-handler/index.js

Comment (ehgmsdk20) [reply event-handler/event-channel-manager.js]: Fixed, thanks!

File: event-handler/state-event-tree-manager.js (hunk 3 above)
Comment (ehgmsdk20) [reply event-handler/state-event-tree-manager.js]: Fixed, thanks!

File: common/constants.js (hunk 1 above)
Comment (ehgmsdk20) [reply common/constants.js]: Fixed, thanks!

**Comment by ehgmsdk20** [comment]
Thanks for review, now merging~

**Pull request merged**